lib/sqlalchemy/dialects/mysql/pyodbc.py

```diff
@@ -40,6 +50,7 @@ def get_lastrowid(self):
 
 
 class MySQLDialect_pyodbc(PyODBCConnector, MySQLDialect):
+    colspecs = util.update_copy(MySQLDialect.colspecs, {TIME: _pyodbcTIME})
```

Comment:
the key here should be the most general type, sqlalchemy.sql.sqltypes.Time  , try that please

Reply:
Yup, that appears to have done the trick.